core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/ParseUtils.java

```diff
@@ -25,7 +25,7 @@
    * @return the index of the first character in toParse from idx that is not a "space.
    */
   public static int skipSpaces(String toParse, int idx) {
-    while (isBlank(toParse.charAt(idx)) && idx < toParse.length()) ++idx;
+    while (idx < toParse.length() && isBlank(toParse.charAt(idx))) ++idx;
```

Comment:
Reported by the IDE, and indeed the order of checks would prevent this method from working in the case where `idx >= toParse.length()`.